lib/worker.sh

```diff
@@ -446,19 +462,58 @@ function TestCodebase() {
         echo -e "${NC}${B[R]}${F[W]}ERROR:${NC} Linter CMD:[$LINTER_COMMAND $FILE]${NC}"
         # Increment the error count
         (("ERRORS_FOUND_$FILE_TYPE++"))
-        # Increment counter that check was ran
-        ((TESTS_RAN++))
       else
         ###########
         # Success #
         ###########
         echo -e "${NC}${F[B]} - File:${F[W]}[$FILE_NAME]${F[B]} failed test case with ${F[W]}[$LINTER_NAME]${F[B]} successfully${NC}"
-        # Increment counter that check was ran
-        ((TESTS_RAN++))
+      fi
+      #######################################################
+      # Store the linting as a temporary file in TAP format #
+      #######################################################
+      if IsTAP ; then
+        echo "not ok ${TESTS_RAN} - ${FILE_NAME}" >> "${TMPFILE}"
+        ##########################################
+        # Report the detailed message if enabled #
+        ##########################################
+        DETAILED_MSG=$(TransformTAPDetails "$LINT_CMD")
+        if [ -n "${DETAILED_MSG}" ] ; then
+          printf "  ---\n  message: %s\n  ...\n" "${DETAILED_MSG}" >> "${TMPFILE}"
+        fi
       fi
     fi
   done
 
+  ###########################################################################
+  # Generate report in TAP format and validate with the expected TAP output #
+  ###########################################################################
+  if IsTAP && [ ${TESTS_RAN} -gt 0 ] ; then
+    printf "TAP version 13\n1..%s\n" "${TESTS_RAN}" > "${REPORT_OUTPUT_FILE}"
+    cat "${TMPFILE}" >> "${REPORT_OUTPUT_FILE}"
+
+    ########################################################################
+    # If expected TAP report exists then compare with the generated report #
+    ########################################################################
+    EXPECTED_FILE="$GITHUB_WORKSPACE/$TEST_CASE_FOLDER/$INDVIDUAL_TEST_FOLDER/reports/expected-${FILE_TYPE}.tap"
+    if [ -e "$EXPECTED_FILE" ] ; then
+      TMPFILE=$(mktemp -q "/tmp/diff-${FILE_TYPE}.XXXXXX")
+
+      ## Ignore white spaces, case sensitive and output only whether files differ
+      if diff -a -w -q -i "${EXPECTED_FILE}" "${REPORT_OUTPUT_FILE}" > "${TMPFILE}" ; then
+        #############################################
+        # We failed to compare the reporting output #
+        #############################################
+        echo -e "${NC}${B[R]}${F[W]}ERROR!${NC} Failed to assert TAP output:[$LINTER_NAME]${NC}"!
+        echo "Please validate the asserts!"
+        cat "${TMPFILE}"
```

Comment:
This might help to understand why it failed by providing the diff output

Reply:
much needed, in case someone updates a test